test/ScheduledChannelTest.cpp

```diff
@@ -0,0 +1,112 @@
+#include "pipert/ScheduledChannel.h"
+#include "pipert/Scheduler.h"
+#include "gtest/gtest.h"
+
+namespace{
+    class OtherPrinter {
+        public:
+        void Print(pipert::PacketToProcess<int> packet) {
+            std::cout << "Other: " << packet.data() << std::endl;
+        }
+        void Add(pipert::PacketToProcess<int> packet) {
+            std::cout << packet.data() << std::endl;
+            packet.data() = 11;
+        }
+    };
+
+} //namespace
+
+class ScheduledChannelTest : public ::testing::Test {
+    protected:
+    ::pipert::Scheduler scheduler_;
+    ::pipert::Timer::Time default_time_;
+    ::pipert::Timer::Time old_time_;
+    ::pipert::Timer::Time new_time_;
+    OtherPrinter op_;
+
+    ScheduledChannelTest() 
+        : scheduler_(1),
+        default_time_(1603716824924000),
+        old_time_(1603716857718209),
+        new_time_(1603716871368358) {
+    }
+
+    virtual ~ScheduledChannelTest() {
+    }
+
+    virtual void SetUp() {
+    }
+
+    virtual void TearDown() {
+    }
+
+};
+
+
+TEST_F(ScheduledChannelTest, ScheduledChannelCreationWithSchedulerBasicChannelInfoTest) {
+    // When
+    ::pipert::ScheduledChannel<int> channel = scheduler_.CreateScheduledChannel<int>("TestChannel", 2, nullptr, std::bind(&OtherPrinter::Print, &op_, std::placeholders::_1));
+    // Then
+    EXPECT_EQ(channel.GetName(), "TestChannel");
+    EXPECT_EQ(channel.GetCapacity(), 2);
+    EXPECT_EQ(channel.GetPacketSize(), sizeof(::pipert::Packet<int>));
+}
+
+TEST_F(ScheduledChannelTest, ScheduledChannelCreationWithSchedulerBufferInfoTest) {
+    // When
+    ::pipert::ScheduledChannel<int> channel = scheduler_.CreateScheduledChannel<int>("TestChannel", 2, nullptr, std::bind(&OtherPrinter::Print, &op_, std::placeholders::_1));
+    // Then
+    EXPECT_EQ(channel.GetFreeBufferLength(), 2);
+    EXPECT_EQ(channel.GetQueuedBufferLength(), 0);
+}
+
+TEST_F(ScheduledChannelTest, ScheduledChannelAcquireTest) {
+    // Given
+    const int value = 42;
+    ::pipert::ScheduledChannel<int> channel = scheduler_.CreateScheduledChannel<int>("TestChannel", 2, nullptr, std::bind(&OtherPrinter::Print, &op_, std::placeholders::_1));
+    // When
+    ::pipert::PacketToFill<int> packet = channel.Acquire("TestChannel", default_time_, value);
```

Comment:
This works fine, but shows a bad example. If you are about to send a packet to the "TestChannel" channel, you are probably not the part of the system called "TestChannel". You are probably in a different Node. So just for showing a good example, let's call yourself, as the client of the acquisiton like "DataSupplier" or something. This is also true for the rest of the tests.